text/0002-poet2-consensus.md

```diff
@@ -0,0 +1,582 @@
+       the winner.
+
+3. If the Chain Head has been updated, discard any existing Proposed Block and
+   start building a new Proposed Block.
+
+## Bootstrapping new validator nodes
+
+When a new validator joins the network, it submits a registration transaction.
+It then starts synchronizing its ledger and state with its neighbors by requesting
+blocks. 
+
+It then performs the following steps to bootstrap its Wall Clock:
+
+1. Start accepting incoming blocks and requesting missing blocks from neighbors 
+	to build the chain until the genesis block is received.
+2. When a block containing the validator's registration transaction is received,
+	start the 'C' block counter. Let this block be known as the 'registration' 
+	block.
+3. Pick a random block number between the 'registration' block and the C'th block.
+	We call this block the 'Synchronization' block, since it will be used to 
+	initialize and synchronize the two clocks. 
+4. To be considered a Synchronization block, the block must extend the current 
+	chain head.
+5. Once the Synchronization block is received, the WallClock & ChainClock are 
+	initialized. This block is henceforth used for ChainClock computation.
+6. Any incoming blocks before the C'th block are considered Eligible.
+7. The start of the validator's participation in consensus (after the C-block
+	delay) is also when the validator starts enforcing the Eligibility
+	checks on incoming blocks.
+
+This mechanism bootstraps the Wall Clock to a reasonable value and allows
+the validator to effectively participate in consensus. 
+
+## Clock Drifts and Network Latency
+
+PoET 2.0 relies on the system clock in two instances: for setting a timer to
+wait for `WaitCertificate.WaitTime` seconds and for maintaining the Wall Clock.
+
+Clock drifts on a validator may cause it to broadcast a block earlier or later
+than intended. On the peer validator handling incoming blocks, clock drifts will
+impact the Wall Clock being used to determine block eligibility. Depending on 
+the drift, blocks may become eligible earlier or later than ideal.
+
+The Leader Election mechanism itself is independent of the clock and will not be
+impacted by drifts.
+
+For forward clock drifts, the primary impact on the network will be of blocks
+being broadcast earlier. This could result in situations where multiple Claim Blocks
+are active at any given time, resulting in decreased efficiency of the consensus 
+mechanism. Negative clock drifts will cause blocks to be broadcast later, potentially
+impacting SLAs for block publishing intervals.
+
+Because Leader Election is isolated from clock drifts, the the clocks need only 
+be reasonably accurate to keep the network operating at the desired level of 
+efficiency. Existing system clock synchronization mechanisms like NTP
+etc. may be sufficient for PoET 2.0 requirements.
+
+Network latencies may be exploited by malicious nodes to broadcast blocks

```

Comment:
Some of the points in this section sound like they could be moved to/or repeated in the currently empty drawbacks section.